sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/CPKTest.groovy

```diff
@@ -0,0 +1,244 @@
+
+        then:
+        response.statusCode() == 201
+        response.value().isServerEncrypted()
+        response.value().encryptionKeySha256() == key.keySHA256()
+    }
+
+    def "Append block with CPK"() {
+        setup:
+        cpkAppendBlob.create()
+
+        when:
+        def response = cpkAppendBlob.appendBlockWithResponse(defaultInputStream.get(), defaultDataSize, null, null, null)
+
+        then:
+        response.statusCode() == 201
+        response.value().isServerEncrypted()
+        response.value().encryptionKeySha256() == key.keySHA256()
+    }
+
+    def "Append block from URL with CPK"() {
+        setup:
+        cpkAppendBlob.create()
+        def sourceBlob = cc.getBlockBlobClient(generateBlobName())
+        sourceBlob.upload(defaultInputStream.get(), defaultDataSize)
+
+        when:
+        def response = cpkAppendBlob.appendBlockFromUrlWithResponse(
+            new URL(sourceBlob.getBlobUrl().toString() + "?" + sourceBlob.generateSAS(OffsetDateTime.now().plusHours(1), new BlobSASPermission().read(true))),
+            null, null, null, null, null, null)
+
+        then:
+        response.statusCode() == 201
+        //TODO uncomment when swagger is fixed so AppendBlobAppendBLockFromURLHeaders contains isrequestserverencrypted
+        //response.value().isServerEncrypted()
+        response.value().encryptionKeySha256() == key.keySHA256()
+    }
+
+    def "Set blob metadata with CPK"() {
+        setup:
+        def metadata = new Metadata()
+        metadata.put("foo", "bar")
+
+        when:
+        def response = cpkExistingBlob.setMetadataWithResponse(metadata, null, null, null)
+
+        then:
+        response.statusCode() == 200
+        Boolean.parseBoolean(response.headers().value(Constants.HeaderConstants.REQUEST_SERVER_ENCRYPTED))
+        response.headers().value(Constants.HeaderConstants.ENCRYPTION_KEY_SHA256) == key.keySHA256()
+    }
+
+    def "Get blob properties and metadata with CPK"() {
+        when:
+        def response = cpkExistingBlob.getPropertiesWithResponse(null, null, null)
+
+        then:
+        response.statusCode() == 200
+        Boolean.parseBoolean(response.headers().value(Constants.HeaderConstants.SERVER_ENCRYPTED))
+        response.headers().value(Constants.HeaderConstants.ENCRYPTION_KEY_SHA256) == key.keySHA256()
+    }
+
+//    TODO unignore when swagger is resolved with service team
+//    def "Set blob tier with CPK"() {
+//        when:
+//        def response = cpkExistingBlob.setTierWithResponse(AccessTier.COOL, null, null, null)
+//
+//        then:
+//        response.statusCode() == 200
+//        Boolean.parseBoolean(response.headers().value(Constants.HeaderConstants.SERVER_ENCRYPTED))
+//        response.headers().value(Constants.HeaderConstants.ENCRYPTION_KEY_SHA256) == key.keySHA256()
+//    }
+
+    def "Snapshot blob with CPK"() {
+        setup:
+        def metadata = new Metadata()
+        metadata.put("foo", "bar")
```

Comment:
Is this necessary?

Reply:
Yes. We need to apply data to be encrypted, otherwise we are not actually invoking encryption on the service and we won't get headers back.